### PR TITLE
[OCPBUGS-34173]: fix sorting unpack jobs

### PR DESCRIPTION
**Pull request by ankitathomas**

### CODE REVIEW

**Review by perdasilva — DISMISSED** [review]
lgtm

File: pkg/controller/bundle/bundle_unpacker.go [hunk 1]
```diff
@@ -868,6 +868,16 @@ func sortUnpackJobs(jobs []*batchv1.Job, maxRetainedJobs int) (latest *batchv1.J
...
+		// sort multiple jobs without the Failed condition
+		if condI == nil && condJ == nil {
+			return jobs[i].CreationTimestamp.After(jobs[j].CreationTimestamp.Time)
+		}
+		if condI == nil {
+			return false
+		}
+		if condJ == nil {
+			return true
+		}
```
Comment (kevinrizza) [inline pkg/controller/bundle/bundle_unpacker.go:880]: is there a test case we can add to this somehow?

Comment (grokspawn) [reply pkg/controller/bundle/bundle_unpacker.go:880]: +1 
https://github.com/operator-framework/operator-lifecycle-manager/blob/889ae50a66fe36a46a3c756a0ad7b481b73672f1/pkg/controller/bundle/bundle_unpacker_test.go#L1977

Comment (oceanc80) [reply pkg/controller/bundle/bundle_unpacker.go:880]: Test cases here should check for this now: https://github.com/ankitathomas/operator-lifecycle-manager/pull/1

**Review by perdasilva — APPROVED** [review]
lgtm

**Pull request merged**